packages/evolution-frontend/src/components/inputs/InputCheckbox.tsx

```diff
@@ -113,7 +113,7 @@ const InputCheckboxChoice = <CustomSurvey, CustomHousehold, CustomHome, CustomPe
                             />
                         </span>
                     )}
-                    <span>{strLabel}</span>
+                    <span dangerouslySetInnerHTML={{ __html : strLabel ? strLabel : '' }} />
```

Comment:
You are unfortunately right! The type we receive is only the widget specific one, but the widgetConfig has a few more common fields, that includes the `containsHtml`. We'll have to rectify those types. But for now, you can just cast to `any` and get the `containsHtml` like this `(this.props.widgetConfig as any).containsHtml` with a `// FIXME` line above. It will be "safer" as one has to explicitly tell that html is want and can't accidently inject something bad.

Reply:
To my view, this is not safer because it can lead to usage errors(developpers' mistakes).`dangerouslySetInnerHTML` already detects if there is any html to be rendered, so there would be no mistake possible (someone wouldn't write html code by accident). And it would be heavier to maintain since we would have 2 props to manipulate (again, I think it could lead to usage errors).

But if we have to add the `containsHtml `, wouldn't it be better to add it to the `choice` itself instead of the whole `widgetConfig` ?

Reply:
**For the record :** 
We are going to implement the `dangerouslySetHtml` without the `containsHtml` attribut.
Our major concern was about a potential security breach. We were worried about a malicious input from users.

That's why we used the `dompurify` library which prevent any XSS attacks like described on this page : https://victoria2666.medium.com/the-implications-of-dangerouslysetinnerhtml-in-react-64e2d02af66a

Moreover, if at one point in the futur, the field `__html` contained in `dangerouslySetHtml` is subject to users' input (meaning that if the variable passed to `__html` is the result of a user input from the frontend), it will be important to limit and regulate the input field to ensure the program safety.